### PR TITLE
Handle atypical DB failure object

### PR DESCRIPTION
**Pull request by mdeuser**
Results in leaving warm container in a bad state in which all subsequent same-action invocation will fail.

### CODE REVIEW

**Comment by mdeuser** [comment]
Travis failed due to API GW timing issue.  PR #1650 is required before this PR will pass Travis.

File: core/routemgmt/getApi.js [hunk 1]
```diff
@@ -119,11 +119,11 @@ function readApiDocument(cloudantDb, docId, params) {
...
     cloudantDb.get(docId, params, function(error, response) {
       if (!error) {
-        console.log('success', JSON.stringify(response));
+        console.log('DB success', JSON.stringify(response));
         resolve(response);
       } else {
-        console.error("DB error: ",error.name, error.error, error.reason, error.headers.statusCode);
-        if (error.headers.statusCode == 404) {  // When NOT FOUND, return empty results
+        console.error('DB error: '+JSON.stringify(error))
+        if (error.headers && error.headers.statusCode == 404) {  // When NOT FOUND, return empty results
```
Comment (csantanapr) [inline core/routemgmt/getApi.js]: read `statusCode` from error object instead and also use `===` for consistency and jshint.
```
if(error.statusCode === 404) {...
```

For reference here is how cloudant/nano return error 
https://github.com/dscape/nano/blob/master/lib/nano.js#L244

File: core/routemgmt/getApi.js [hunk 2]
```diff
@@ -137,11 +137,11 @@ function readFilteredApiDocument(cloudantDb, designDocId, designDocViewName, par
...
     cloudantDb.view(designDocId, designDocViewName, params, function(error, response) {
       if (!error) {
-        console.log('success', response);
+        console.log('DB view success', JSON.stringify(response));
         resolve(response);
       } else {
-        console.error('error', JSON.stringify(error));
-        if (error.headers.statusCode == 404) {  // When NOT FOUND, return empty results
+        console.error('DB view error', JSON.stringify(error));
+        if (error.headers && error.headers.statusCode == 404) {  // When NOT FOUND, return empty results
```
Comment (csantanapr) [inline core/routemgmt/getApi.js]: same as above use `error.statusCode === 404`

**Comment by mdeuser** [comment]
PG1 1061

**Comment by mdeuser** [comment]
@csantanapr - Ready for re-review/merge.  PG failure due to 2 other unrelated known issues.

File: core/routemgmt/getApi.js [hunk 3]
```diff
@@ -119,11 +119,11 @@ function readApiDocument(cloudantDb, docId, params) {
...
     cloudantDb.get(docId, params, function(error, response) {
       if (!error) {
-        console.log('success', JSON.stringify(response));
+        console.log('DB success', JSON.stringify(response));
         resolve(response);
       } else {
-        console.error("DB error: ",error.name, error.error, error.reason, error.headers.statusCode);
-        if (error.headers.statusCode == 404) {  // When NOT FOUND, return empty results
+        console.error('DB error: '+JSON.stringify(error))
+        if (error.headers && error.headers.statusCode === 404) {  // When NOT FOUND, return empty results
```
Comment (csantanapr) [inline core/routemgmt/getApi.js]: Why not use `if(error.statusCode === 404) {`

File: core/routemgmt/getApi.js [hunk 4]
```diff
@@ -137,11 +137,11 @@ function readFilteredApiDocument(cloudantDb, designDocId, designDocViewName, par
...
     cloudantDb.view(designDocId, designDocViewName, params, function(error, response) {
       if (!error) {
-        console.log('success', response);
+        console.log('DB view success', JSON.stringify(response));
         resolve(response);
       } else {
-        console.error('error', JSON.stringify(error));
-        if (error.headers.statusCode == 404) {  // When NOT FOUND, return empty results
+        console.error('DB view error', JSON.stringify(error));
+        if (error.headers && error.headers.statusCode === 404) {  // When NOT FOUND, return empty results
```
Comment (csantanapr) [inline core/routemgmt/getApi.js]: same comment as above, Why not use `if(error.statusCode === 404) {`

Comment (mdeuser) [inline core/routemgmt/getApi.js]: based on the nano.js code, it seems either are acceptable.  i found examples using both statusCode values... but i think using the top level is a bit more readable

**Comment by mdeuser** [comment]
PG3 133.  All API GW tests passed.

**Comment by csantanapr** [comment]
rebase and merged 4a2b2455ddb144fd3ff921dd2d78146ccae7acf3

**Pull request closed without merging**